### PR TITLE
Add LoadCollection instant loading interface

### LINKED ISSUE #29: Implement LoadCollection() interface
This api depends on ShowCollections()

### PR DESCRIPTION
**Pull request by ArkaprabhaChakraborty**
fixes #29 
Signed-off-by: ArkaprabhaChakraborty <chakrabortyarkaprabha998@gmail.com>

### CODE REVIEW

**Comment by ArkaprabhaChakraborty** [comment]
@yhmo @matrixji I need some help on how to use the progress monitor here. Some hints will do.

**Comment by matrixji** [comment]
> @yhmo @matrixji I need some help on how to use the progress monitor here. Some hints will do.
Seems LoadPartitions are very similar to LoadCollection, you could check it for examples

**Comment by ArkaprabhaChakraborty** [comment]
@matrixji can you please tell me what im doing wrong?

**Comment by matrixji** [comment]
> @matrixji can you please tell me what im doing wrong?

The ShowCollection was called in LoadCollection's wait part, so you should add EXPECT_CALL(ShowCollection) in the test code.

You could search the test code `TEST_F(MilvusMockedTest, LoadPartitionsWithQueryStatusSuccess) ` for references.

**Comment by yhmo** [comment]
MilvusClientImpl::waitForStatus() is called in the wait_for_status. ShowCollections() is called to check loading progress, but you didn't specify 100 percent, the waitForStatus() will repeatly check until time out, finally return  StatusCode::TIMEOUT. Since you set 5 seconds timeout for the ProgressMonitor, so the test case will be failed after 5 ~ 6 seconds.

You need to specify 100 memory percentage before you call the DescribeCollection():
```
EXPECT_CALL(service_,
                ShowCollections(_,_,_))
        .Times(1)
        .WillRepeatedly(
            [&partitions](::grpc::ServerContext*, const ShowCollectionsRequest*, ShowCollectionsResponse* response) {
               // set memory percentage to be 100% here
                return ::grpc::Status{};
            });
```

**Comment by yhmo** [comment]
@ArkaprabhaChakraborty 
There are 4 commits in this pr, please quash them to be one, just follow this doc: https://www.git-tower.com/learn/git/faq/git-squash

File: test/it/TestLoadCollection.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        .WillOnce([](::grpc::ServerContext*, const LoadCollectionRequest*, milvus::proto::common::Status*) {
+            return ::grpc::Status{};
+        });
+
+    EXPECT_CALL(service_, ShowCollections(_, _, _))
+        .Times(10)
+        .WillRepeatedly([&](::grpc::ServerContext*, const ShowCollectionsRequest*, ShowCollectionsResponse* response) {
+            auto n = response->inmemory_percentages_size();
+            for (int i = 0; i < n; ++i) {
+                response->set_inmemory_percentages(i, 100);
```
Comment (matrixji) [inline test/it/TestLoadCollection.cpp:64]: Here will return all percentages to 100 the first time, but you want the `ShowCollections` called ten times.

Comment (ArkaprabhaChakraborty) [reply test/it/TestLoadCollection.cpp:64]: So Times(10) should be set?

Comment (matrixji) [reply test/it/TestLoadCollection.cpp:64]: .`WillOnce` for `ShowCollections` or set percentages10, 20, 30, 40 .... 100 in the mock of ShowCollections.

**Comment by ArkaprabhaChakraborty** [comment]
cc: @matrixji @yhmo

**Pull request merged**